src/docs/release/breaking-changes/text-field-material-localizations.md

```diff
@@ -0,0 +1,122 @@
+description: TextFields will now assert for Material Localizations.
+---
+
+## Summary
+
+Instances of `TextField` must have a `MaterialLocalizations` present.
+Trying to instantiate  a `TextField` without the proper localizations will
+result in an assertion such as the following:
+
+```
+No MaterialLocalizations found.
+TextField widgets require MaterialLocalizations to be provided by a Localizations widget ancestor.
+Localizations are used to generate many different messages, labels, and abbreviations which are used
+by the material library.
+To introduce a MaterialLocalizations, either use a MaterialApp at the root of your application to
+include them automatically, or add a Localization widget with a MaterialLocalizations delegate.
+The specific widget that could not find a MaterialLocalizations ancestor was:
+  TextField
+```
+
+## Context
+
+If your `TextField` is within the scope of a `MaterialApp`, the
+`DefaultMaterialLocalizations` is already be instantiated and may suit your needs without having
+to make a change to your existing code.
+
+If your `TextField` is not within a `MaterialApp`, you may provide the localizations of
+your choosing using the `Localizations` widget.
+
+## Migration guide
+
+If you are seeing an assertion error, make sure locale information is being
+provided to your `TextField` through an ancestor `MaterialApp`, which will automatically provide
+`MaterialLocalizations`, or with your own `Localizations` widget.
+
+Code before migration:
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MediaQuery(
+      data: const MediaQueryData(),
+      child: Directionality(
+        textDirection: TextDirection.ltr,
+        child: TextField(),
+      ),
+    );
+  }
+}
+```
+
+Code after migration (Providing localizations via the `MaterialApp`):
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: TextField(),
+    );
+  }
+}
+```
+
+Code after migration (Providing localizations via the `Localizations` widget):
+
+```dart
+import 'package:flutter/material.dart';
+void main() => runApp(Foo());
+class Foo extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Localizations(
+      locale: const Locale('en', 'US'),
+      delegates: <LocalizationsDelegate<dynamic>>[
+        DefaultWidgetsLocalizations.delegate,
+        DefaultMaterialLocalizations.delegate,
+      ],
+      child: MediaQuery(
+        data: const MediaQueryData(),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: TextField(),
+        ),
+      ),
+    );
+  }
+}
+```
+
+## Timeline
+
+TBD - Add the version number that contains this change.
```

Comment:
I can't land this PR until we have actual release info.

Reply:
I am aware.